www/src/components/navigation/tableOfContents.astro

```diff
@@ -50,19 +50,66 @@ headings = [{ depth: 2, slug: "overview", text: title }, ...headings].filter(
       (entries) => {
         entries.forEach((entry) => {
           const id = entry.target.id;
+          // do nothing if id is toc-heading
+          if (id === "toc-heading") return;
           // have to do "All" here cause we render this component in pageContent for mobile too
           const tocItem = Array.from(
             document.querySelectorAll(`a[href="#${id}"]`),
           ).at(-1);
           // this occurs when the id = "toc-heading"
           if (!tocItem) return;
```

Comment:
do we still need this line with the check you added in line 54?

Reply:
yeah we can remove it same as line 58 check